kolibri/plugins/learn/assets/src/views/content-card-group-grid/index.vue

```diff
@@ -22,10 +22,12 @@
     <content-card
       v-for="content in contents"
       v-show="showContentCard(content)"
+      class="grid-item"
+      :isMobile="isMobile"
+      :class="{ 'mobile': isMobile }"
```

Comment:
Instead of attaching this class here, can you just apply the styling for this `mobile` class inside the content-card?

Reply:
I sure can, but what's the reasoning?

Reply:
If I can, I try to avoid attaching classes to components.

Reply:
Is that to avoid collisions between styling set by the parent component and the child component? If so, I agree entirely.

I did however use this pattern in a couple other places:

* https://github.com/learningequality/kolibri/pull/2488/files#diff-11803d4dbb8658106955f132e60141a5R5
* https://github.com/learningequality/kolibri/pull/2488/files#diff-34353620ee4080ac1c253020c24181afR3

Mainly because it enables styling like this:

* https://github.com/learningequality/kolibri/pull/2488/files#diff-34353620ee4080ac1c253020c24181afR98
* https://github.com/learningequality/kolibri/pull/2488/files#diff-11803d4dbb8658106955f132e60141a5R184

Trying to think how to achieve both .. for now I'll do as you suggest

Reply:
oh, I misunderstood ... you're saying not to attach classes to components from the parent? I don't understand why?

Reply:
(turns out this styling is redundant with what's being set inside anyway - deleting)